File: paralelismo/matrix_mult_ventilator_worker_sink/ejemplo_profesor/worker.py
import sys
import time
import zmq

context = zmq.Context()

work = context.socket(zmq.PULL)
work.connect("tcp://localhost:5557")

# Socket to send messages to
sink = context.socket(zmq.PUSH)
sink.connect("tcp://localhost:5558")

# Process tasks forever
tasks = 0
while True:
    s = work.recv()
    tasks += 1
    # Do the work
    time.sleep(int(s)*0.001)
 
    # Send results to sink
    sink.send(b'')

    print(f"I do {tasks} tasks")

File: paralelismo/matrix_mult_ventilator_worker_sink/ejemplo_profesor/worker.cc
#include <iostream>
#include <zmqpp/zmqpp.hpp>
#include <string>
#include <chrono>
#include <thread>

using namespace std;
using namespace zmqpp;

int main() {
  context ctx;
  socket work(ctx, socket_type::pull);
  work.connect("tcp://localhost:5557");

  socket sink(ctx, socket_type::push);
  sink.connect("tcp://localhost:5558");

  while(true) {
    message s;
    work.receive(s);

    string text;
    s >> text;
    cout << "Work received " << text << endl;

    int tsleep = stoi(text);
    this_thread::sleep_for(chrono::milliseconds(tsleep));

    message rep;
    rep << "hi!";
    sink.send(rep);
  }


  cout << "Hello from C++ folks" << endl;
  return 0;
}
